File: 백준/Silver/11723. 집합/집합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	
	public static void main(String[] args) throws IOException {
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st;
		StringBuilder sb = new StringBuilder();
		
		boolean[] arr = new boolean[21];
		
		int N = Integer.parseInt(br.readLine());
		for (int i=0; i<N; i++) {
			st = new StringTokenizer(br.readLine());
			String op = st.nextToken();
			int x = 0;
			switch (op) {
			case "add" :
				x = Integer.parseInt(st.nextToken());
				arr[x] = true;
				break;
			case "remove" :
				x = Integer.parseInt(st.nextToken());
				arr[x] = false;
				break;
			case "check" :
				x = Integer.parseInt(st.nextToken());
				if (arr[x]) {
					sb.append("1");
				} else {
					sb.append("0");
				}
				sb.append("\n");
				break;
			case "toggle" :
				x = Integer.parseInt(st.nextToken());
				arr[x] = !arr[x];
				break;
			case "all" :
				Arrays.fill(arr, true);
				break;
			case "empty" :
				Arrays.fill(arr, false);
			}
		}
		System.out.println(sb.toString());
		
	}
    
}

File: 백준/Silver/11723. 집합/집합.py
import sys
input = sys.stdin.readline

S = [False]*21
N = int(input())
for _ in range(N):
    op = input().split()
    if op[0] == "add" and not S[int(op[1])]:
        S[int(op[1])] = True
    elif op[0] == "remove" and S[int(op[1])]:
        S[int(op[1])] = False
    elif op[0] == "check":
        if S[int(op[1])]:
            print(1)
        else:
            print(0)
    elif op[0] == "toggle":
        S[int(op[1])] = not S[int(op[1])]
    elif op[0] == "all":
        for i in range(1, 21):
            S[i] = True
    elif op[0] == "empty":
        for i in range(1, 21):
            S[i] = False
    else:
        pass
        # 위에서 add, remove의 조건에 의해 아무것도 수행되지 않는 경우 이 라인이 수행
        # 따라서 empty를 else로 처리하면 문제가 생김
